packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -420,8 +441,12 @@ class AndroidDevice extends Device {
       return false;
     }
     if (installResult.exitCode != 0) {
-      _logger.printError('Error: ADB exited with exit code ${installResult.exitCode}');
-      _logger.printError('$installResult');
+      if (installResult.stderr.contains('Bad user number')) {
```

Comment:
We could also run `adb shell pm list users` for them to list possible users

Reply:
We could it for them, but I don't want to parse the output since there's no machine readable format options and this I doubt this flag will be used except by users who are already familiar with Android and `adb` (see b/142944798).
```
$ adb shell pm list users       

Users:
	UserInfo{0:Owner:13} running
	UserInfo{10:jane:10} running
```
I don't think launching will work if the user isn't running (though install will?) and I don't want particularly want to explain all that.

Reply:
Makes sense!